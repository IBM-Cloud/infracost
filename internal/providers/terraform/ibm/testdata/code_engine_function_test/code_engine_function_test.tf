resource "ibm_resource_group" "test_group" {
  name = "test-resource-group"
}

resource "ibm_code_engine_project" "ce_project" {
  name              = "ce_project"
  resource_group_id = ibm_resource_group.test_group.id
}

resource "ibm_code_engine_function" "ce_function" {
  project_id                    = ibm_code_engine_project.ce_project.id
  name                          = "ce_function"
  code_reference                = "icr.io/codeengine/helloworld"
  scale_memory_limit            = "4G"
  scale_cpu_limit               = "1"
  runtime                       = "nodejs-20"
}

resource "ibm_code_engine_function" "ce_function2" {
  project_id                    = ibm_code_engine_project.ce_project.id
  name                          = "ce_function2"
  code_reference                = "icr.io/codeengine/helloworld"
  scale_memory_limit            = "2G"
  scale_cpu_limit               = "0.5"
  runtime                       = "nodejs-20"
}
