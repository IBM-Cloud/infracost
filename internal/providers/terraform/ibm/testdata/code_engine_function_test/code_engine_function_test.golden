
 Name                                   Monthly Qty  Unit                Monthly Cost 
                                                                                      
 ibm_code_engine_function.ce_function                                                 
 ├─ Virtual Processor Cores                       1  vCPU Hours                 $0.12 
 ├─ RAM                                           4  GB Hours                   $0.05 
 └─ Million HTTP calls                       0.0001  Million HTTP calls         $0.00 
                                                                                      
 ibm_code_engine_function.ce_function2                                                
 ├─ Virtual Processor Cores                     0.5  vCPU Hours                 $0.06 
 ├─ RAM                                           2  GB Hours                   $0.03 
 └─ Million HTTP calls                       0.0006  Million HTTP calls         $0.00 
                                                                                      
 OVERALL TOTAL                                                                  $0.26 
──────────────────────────────────
4 cloud resources were detected:
∙ 2 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 2 were free:
  ∙ 1 x ibm_code_engine_project
  ∙ 1 x ibm_resource_group